2006-05-12  Jakub Jelinek  <jakub@redhat.com>

	* crtstuff.c (__dso_handle): Make __dso_handle const.

--- gcc/crtstuff.c.jj	2005-11-16 10:02:51.000000000 +0100
+++ gcc/crtstuff.c	2006-05-12 18:28:18.000000000 +0200
@@ -226,12 +226,12 @@ STATIC void *__JCR_LIST__[]
    dynamic linker takes care of this.  */
 
 #ifdef HAVE_GAS_HIDDEN
-extern void *__dso_handle __attribute__ ((__visibility__ ("hidden")));
+extern void *const __dso_handle __attribute__ ((__visibility__ ("hidden")));
 #endif
 #ifdef CRTSTUFFS_O
-void *__dso_handle = &__dso_handle;
+void *const __dso_handle = (void *) &__dso_handle;
 #else
-void *__dso_handle = 0;
+void *const __dso_handle = 0;
 #endif
 
 /* The __cxa_finalize function may not be available so we use only a
